internal/services/configstore/api/maintenance.go

```diff
@@ -21,8 +21,35 @@ import (
 	"agola.io/agola/internal/util"
 
 	"github.com/rs/zerolog"
+
+	csapitypes "agola.io/agola/services/configstore/api/types"
 )
 
+type MaintenanceStatusHandler struct {
+	log zerolog.Logger
+	ah  *action.ActionHandler
+}
+
+func NewMaintenanceStatusHandler(log zerolog.Logger, ah *action.ActionHandler) *MaintenanceStatusHandler {
+	return &MaintenanceStatusHandler{log: log, ah: ah}
+}
+
+func (h *MaintenanceStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	enabled, err := h.ah.IsMaintenanceEnabled(ctx)
```

Comment:
This will only check if maintenance mode has been set to be enabled but not that it was already enabled since this is an asynchronous task. You should report two statuses: requested status and current status. The current status is enabled when the maintenance router is enabled.